src/Loop/Driver.php

```diff
@@ -285,4 +303,56 @@
      * @return null|object|resource The loop handle the event loop operates on. `null` if there is none.
      */
     abstract public function getHandle();
+
+    /**
+     * Create a watcher identifier.
+     *
+     * Any watcher identifiers returned to clients by the driver MUST be created using this method. The returned watcher
+     * identifier SHALL be unique for the duration of the PHP process.
+     *
+     * @return string The watcher identifier.
+     */
+    final protected function createWatcherIdentifier()
+    {
+        if (!isset($this->driverId)) {
+            $this->driverId = self::$nextDriverId++;
```

Comment:
I see no problem with that. It's expected when you're extending a class with a constructor.